src/H5Tconv.c

```diff
@@ -3006,28 +3119,28 @@ H5T__conv_enum(H5T_t *src, H5T_t *dst, H5T_cdata_t *cdata, const H5T_conv_ctx_t
                                 conv_ctx->u.conv.dst_type_id, s, d, conv_ctx->u.conv.cb_struct.user_data);
```

Comment:
I believe that this can underflow as well as overflow, in which case it should throw a RANGE_LO exception.

Reply:
I wouldn't be surprised if there are bugs with conversions on enums. There are old issues still open specifically around enums. I was trying to avoid making any changes to functionality.